master/internal/db/postgres_task_config_policy.go

```diff
@@ -0,0 +1,163 @@
+}
+
+// AddNTSCConfigPolicies adds the NTSC invariant config and constraints config policies to
+// the database.
+func AddNTSCConfigPolicies(ctx context.Context,
+	experimentTCP *model.ExperimentTaskConfigPolicy) error {
+	return Bun().RunInTx(ctx, nil, func(ctx context.Context, tx bun.Tx) error {
+		return AddNTSCConfigPoliciesTx(ctx, &tx, experimentTCP)
+	})
+}
+
+// AddNTSCConfigPoliciesTx adds the NTSC invariant config and constraints config policies to
+// the database.
+func AddNTSCConfigPoliciesTx(ctx context.Context, tx *bun.Tx,
+	ntscTCP *model.ExperimentTaskConfigPolicy) error {
+
+	// Validate NTSC invariant config and constraints.
+	expInvariantConfig, err := json.Marshal(ntscTCP.InvariantConfig)
+	if err != nil {
+		return errors.Wrapf(err, "error handling experiment invariant config %v",
+			expInvariantConfig)
+	}
+
+	expConstraints, err := json.Marshal(ntscTCP.Constraints)
+	if err != nil {
+		return errors.Wrapf(err, "error handling experiment constarints %v",
+			expConstraints)
+	}
+
+	// Insert invariant configs and constraints.
+	_, err = Bun().NewInsert().Model(ntscTCP).Exec(ctx)
+	if err != nil {
+		return fmt.Errorf("error inserting NTSC task config policy: %w", err)
+	}
+	return err
+}
+
+// GetExperimentConfigPolicies retrieves the invariant experiment config and constraints for the
+// given scope (global or workspace-level).
+func GetExperimentConfigPolicies(ctx context.Context,
+	scope *int) (*model.ExperimentTaskConfigPolicy, error) {
+	experimentTCP, _, err := getConfigPolicies(ctx, scope, model.ExperimentType)
+	if err != nil {
+		return nil, err
+	}
+	return experimentTCP, nil
+}
+
+func getConfigPolicies(ctx context.Context,
+	scope *int, workloadType model.WorkloadType) (*model.ExperimentTaskConfigPolicy,
+	*model.NTSCTaskConfigPolicy, error) {
+	var experimentTCP model.ExperimentTaskConfigPolicy
+	var ntscTCP model.NTSCTaskConfigPolicy
+
+	switch workloadType {
+	case model.ExperimentType:
+		err := Bun().NewSelect().Model(&experimentTCP).Where("id = ? AND workload_type = ?", scope,
+			workloadType.String())
+		if err != nil {
+			if scope == nil {
+				return nil, nil, fmt.Errorf("error retrieving global experiment task config "+
+					"policies: %w", err)
+			}
+			return nil, nil, fmt.Errorf("error retrieving experiment task config policies for "+
+				"workspace with ID %d: %w", *scope, err)
+		}
+	case model.NTSCType:
+		err := Bun().NewSelect().Model(&ntscTCP).Where("id = ? AND workload_type = ?", scope,
+			workloadType.String())
+		if err != nil {
+			if scope == nil {
+				return nil, nil, fmt.Errorf("error retrieving global NTSC task config "+
+					"policies: %w", err)
+			}
+			return nil, nil, fmt.Errorf("error retrieving NTSC task config policies for "+
+				"workspace with ID %d: %w", *scope, err)
+		}
+	default:
+		return nil, nil, status.Errorf(codes.InvalidArgument,
+			"invalid workload type for config policy: %s", workloadType.String())
+	}
+	return &experimentTCP, &ntscTCP, nil
```

Comment:
When returning, one of `experimentTCP` or `ntscTCP` will be `nil`? Both `GetNTSCConfigPolicies` / `GetExperimentConfigPolicies` call this function; what is the advantage of having `getConfigPolicies` as a separate function over moving the `case model.ExperimentType` to the `GetExperimentConfigPolicies` function?

Reply:
Greaatt question 😂! I initially created the helper to add some clean abstraction and reduce code duplication, but somewhere along the lines of realizing I couldn't, it just turned into a switch statement 🙈
I just removed this helper and put the code from the switch statements back into their respective functions